idb/postgres/internal/writer/writer_test.go

```diff
@@ -1294,3 +1332,45 @@ func TestWriterAddBlockTwice(t *testing.T) {
 	err = pgutil.TxWithRetry(db, serializable, f, nil)
 	require.NoError(t, err)
 }
+
+func TestWriterAddBlockInnerTxnsAssetCreate(t *testing.T) {
+	db, shutdownFunc := setupPostgres(t)
+	defer shutdownFunc()
+
+	// App call with inner txns, should be intra 0, 1, 2
+	appCall := test.MakeAppCallWithInner(test.AccountA, test.AccountB, test.AccountC)
+
+	// Asset create call, should have intra = 3
+	assetCreate := test.MakeAssetConfigTxn(
+		0, 100, 1, false, "ma", "myasset", "myasset.com", test.AccountA)
+
+	block, err := test.MakeBlockForTxns(test.MakeGenesisBlock().BlockHeader, &appCall, &assetCreate)
+	require.NoError(t, err)
+
+	err = makeTx(db, func(tx pgx.Tx) error {
+		w, err := writer.MakeWriter(tx)
+		require.NoError(t, err)
+		defer w.Close()
+
+		err = w.AddBlock(&block, block.Payset, ledgercore.StateDelta{})
+		require.NoError(t, err)
+
+		return tx.Commit(context.Background())
+	})
+
+	txns, err := txnQuery(db, "SELECT * FROM txn ORDER BY intra")
+	require.NoError(t, err)
+	require.Len(t, txns, 2)
+
+	// Verify that intra is correctly assigned
+	require.Equal(t, 0, txns[0].intra, "Intra should be assigned 0 - 3.")
```

Comment:
I think the message is a bit confusing.

Reply:
I think it makes more sense in the followup PR where I check the outer and inner transaction intras. I'll change it here anyway.